polaris/dataset/_adapters.py

```diff
@@ -0,0 +1,22 @@
+from enum import Enum
+
+import datamol as dm
+
+
+class Adapter(Enum):
+    """
+    Adapters are predefined callables that change the format of the data.
+    Adapters are serializable and can thus be saved alongside datasets.
+
+    Attributes:
+        SMILES_TO_MOL: Convert a SMILES string to a RDKit molecule.
+        BYTES_TO_MOL: Convert a RDKit binary string to a RDKit molecule.
+    """
```

Comment:
I can imagine extending the adaptors with `SMARTS_TO_MOL` etc.

Reply:
For sure! More generally, I think we could add all sorts of adapters. To keep things simple, however, I think we should add these as they are needed / requested.